src/jcstress/java/reactor/pool/SimpleDequePoolStressTest.java

```diff
@@ -0,0 +1,105 @@
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.pool;
+
+import java.time.Duration;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.openjdk.jcstress.annotations.Actor;
+import org.openjdk.jcstress.annotations.Arbiter;
+import org.openjdk.jcstress.annotations.JCStressTest;
+import org.openjdk.jcstress.annotations.Outcome;
+import org.openjdk.jcstress.annotations.State;
+import org.openjdk.jcstress.infra.results.IZI_Result;
+import org.openjdk.jcstress.infra.results.IZL_Result;
+import org.openjdk.jcstress.infra.results.IZ_Result;
+import org.openjdk.jcstress.infra.results.I_Result;
+import org.openjdk.jcstress.infra.results.LI_Result;
+import org.openjdk.jcstress.infra.results.LLI_Result;
+import org.openjdk.jcstress.infra.results.LLLL_Result;
+import org.openjdk.jcstress.infra.results.LLL_Result;
+import org.openjdk.jcstress.infra.results.ZZ_Result;
+
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import static org.openjdk.jcstress.annotations.Expect.*;
+
+public class SimpleDequePoolStressTest {
+
+	@JCStressTest
+	@Outcome(id = {"ACQUIRED 1, 0"}, expect = ACCEPTABLE,  desc = "acquired")
+	@Outcome(id = {"NOT ACQUIRED, 1"}, expect = ACCEPTABLE,  desc = "acquired failed fast during/before destroy")
+	@Outcome(id = {"ACQUIRED 111, 0"}, expect = FORBIDDEN,  desc = "acquired the first resource, before destroy")
+	@Outcome(id = {"ACQUIRED 1111, 0"}, expect = FORBIDDEN,  desc = "acquired the first resource, after destroy")
+	@State
+	public static class MaxPendingAcquireWhileDestroying {
+
+		final AtomicInteger resource = new AtomicInteger(111);
+		final AtomicBoolean firstResourceCreated = new AtomicBoolean();
+
+		final SimpleDequePool<AtomicInteger> pool = PoolBuilder
+				.from(Mono.defer(() -> {
+					if (firstResourceCreated.getAndSet(true)) {
+						return Mono.just(new AtomicInteger(1));
+					}
+					resource.compareAndSet(0, 1);
+					return Mono.just(resource);
+				}))
+				.sizeBetween(0, 1)
+				.maxPendingAcquire(0)
+				.releaseHandler(ai -> Mono.fromRunnable(ai::incrementAndGet))
+				.evictionPredicate((res, meta) -> {
+					int v = res.get();
+					return v > 0 && v != 111;
+				})
+				.destroyHandler(ai -> Mono.fromRunnable(() -> ai.addAndGet(1000)))
+				.build(conf -> new SimpleDequePool<>(conf, true));
+
+		private final PooledRef<AtomicInteger> ref;
+
+		{
```

Comment:
does it really need to be an instance initializer? Doesn't seem to throw, so the field initializer can be used

Reply:
indeed, leftover from a previous iteration